axelor-account/src/main/java/com/axelor/apps/account/service/move/MoveValidateServiceImpl.java

```diff
@@ -473,4 +481,35 @@ private String getPartnerFullName(Partner partner) {
       return "" + partner.getId();
     }
   }
+
+  protected void exceptionInactiveAnalyticJournal(Move move) throws AxelorException {
+    if (move != null && CollectionUtils.isNotEmpty(move.getMoveLineList())) {
+      int inactiveNbr = 0;
+      List<String> inactiveList = new ArrayList();
+      for (MoveLine moveLine : move.getMoveLineList()) {
+        if (CollectionUtils.isNotEmpty(moveLine.getAnalyticMoveLineList())) {
+          for (AnalyticMoveLine analyticMoveLine : moveLine.getAnalyticMoveLineList()) {
+            AnalyticJournal analyticJournal = analyticMoveLine.getAnalyticJournal();
+            if (analyticJournal.getStatusSelect() != null
+                && analyticJournal.getStatusSelect() != AnalyticJournalRepository.STATUS_ACTIVE
+                && !inactiveList.contains(analyticJournal.getName())) {
+              inactiveNbr++;
+              inactiveList.add(analyticJournal.getName());
+            }
+          }
+        }
+        if (inactiveNbr == 1) {
+          throw new AxelorException(
+              TraceBackRepository.CATEGORY_CONFIGURATION_ERROR,
+              I18n.get(IExceptionMessage.INACTIVE_ANALYTIC_JOURNAL_FOUND),
+              inactiveList.get(0));
+        } else if (inactiveNbr > 1) {
+          throw new AxelorException(
+              TraceBackRepository.CATEGORY_CONFIGURATION_ERROR,
+              I18n.get(IExceptionMessage.INACTIVE_ANALYTIC_JOURNALS_FOUND),
+              inactiveList.stream().map(code -> code).collect(Collectors.joining(", ")));
+        }
+      }
```

Comment:
I think using functional style programming here could improve lisibility (using streams, map, `.filter(Objects::nonNull)` and count methods)